### PR TITLE
Add options for not invalidating cache when upsert

### PR DESCRIPTION
**Pull request by samUber**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Changelog
 ## v3.0.1 (2018-06-07)
 - Fixed version const
-
+- Add functionality to not invalidate cache when upsert (#341)
```
Comment (phliar) [inline CHANGELOG.md]: 3.0.1 has already been released, please make a new section for 3.0.2 and put your change there.

```
# Changelog

## v3.0.2 (unreleased)
 - Add functionality to not invalidate cache when upsert (#341)

## v3.0.1 (2018-06-07)
 - Fixed version const
...
```

**Comment by samUber** [comment]
updated the change log

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # Changelog
+## v3.0.2 (unreleased)
+ - Add functionality to not invalidate cache when upsert (#341)
 ## v3.0.1 (2018-06-07)
 - Fixed version const
-
```
Comment (phliar) [inline CHANGELOG.md]: We do want a blank line between the version stanzas

File: connectors/cache/fallback.go [hunk 3]
```diff
@@ -49,12 +49,31 @@ type rangeQuery struct {
...
+
 // NewConnector creates a fallback cache connector
-func NewConnector(origin, fallback dosa.Connector, scope metrics.Scope, entities ...dosa.DomainObject) *Connector {
-	return newConnector(origin, fallback, scope, encoding.NewGobEncoder(), entities...)
+func NewConnector(origin, fallback dosa.Connector, scope metrics.Scope, entities []dosa.DomainObject, options ...Options) *Connector {
+	c := newConnector(origin, fallback, scope, encoding.NewGobEncoder(), entities)
+	for _, option := range options {
+		err := option(c)
+		if err != nil {
+			// It's an no-op when getting an error now
```
Comment (phliar) [inline connectors/cache/fallback.go]: Nit: a no-op

**Pull request merged**